# Disneyland Ride Times Analysis

In [30]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [31]:
%sql mysqldb://USERNAME:PASSWORD@DATABASE_ENDPOINT/DATABASE_NAME

### Exploratory Queries

###### Query 1: Identifying Date Inconsistancies

The purpose of this query is to identify the dates during which the data was collected. This is important because it will allow for more consistent data. Identifying dates also allows us to see which dates were consecutive, giving us more data to analyze data over a longer period of time and results that are more reflective of the average wait time at Disneyland.

In [32]:
%%sql
SELECT 
    rwt.last_updated
FROM ride_wait_times rwt
GROUP BY rwt.last_updated
ORDER BY last_updated;

 * mysql://admin:***@lmu-dev-02.cmrqhkyehmsg.us-west-1.rds.amazonaws.com/sql_project
16 rows affected.


[(datetime.datetime(2022, 4, 7, 0, 0),),
 (datetime.datetime(2022, 4, 8, 0, 0),),
 (datetime.datetime(2022, 4, 22, 0, 0),),
 (datetime.datetime(2022, 4, 23, 0, 0),),
 (datetime.datetime(2022, 4, 24, 0, 0),),
 (datetime.datetime(2022, 4, 25, 0, 0),),
 (datetime.datetime(2022, 4, 26, 0, 0),),
 (datetime.datetime(2022, 4, 27, 0, 0),),
 (datetime.datetime(2022, 4, 28, 0, 0),),
 (datetime.datetime(2022, 4, 29, 0, 0),),
 (datetime.datetime(2022, 4, 30, 0, 0),),
 (datetime.datetime(2022, 5, 1, 0, 0),),
 (datetime.datetime(2022, 5, 2, 0, 0),),
 (datetime.datetime(2022, 5, 3, 0, 0),),
 (datetime.datetime(2022, 5, 4, 0, 0),),
 (datetime.datetime(2022, 5, 5, 0, 0),)]

From this query, we can see that data was collected consecutively between 2022-04-22 and 2022-05-04. However, because data collection was started late for 2022-04-22, it was not until 2022-04-23 that there was a full day of data; similarly, we would need to collect through 2022-05-04 and not 2022-05-05, since 2022-05-05 would not have a full day's worth of data. Therefore, it is important to retrieve data collected between 2022-04-23 and 2022-05-04 in order to analyze data collected full days only. When more data is collected, I would recommend querying results by week in order to determine what the wait time averages are per week, which could inform which rides Disneyland should advertise more the next week.

##### Query 2: Identifying Amount of Data Per Park

In this query, we are going to perform a count on the number of records gathered for Disneyland and Disney California Adventures. This will allow us to see if there are comparable amounts of records for each park. If there are not the same amount of records, then it would be more effective to analyze the park with more data, as it will provide more general insights into the wait times of the park.

In [34]:
%%sql
SELECT 
    p.name,
    COUNT(rwt.wait_time) AS number_of_wait_times
FROM parks p
JOIN rides r
    ON p.park_id = r.park_id
JOIN ride_wait_times rwt
    ON r.id = rwt.id
WHERE rwt.last_updated BETWEEN '2022-04-23' AND '2022-05-04'
GROUP BY p.name
ORDER BY last_updated;

 * mysql://admin:***@lmu-dev-02.cmrqhkyehmsg.us-west-1.rds.amazonaws.com/sql_project
2 rows affected.


[('Disneyland', 14688), ('Disney California Adventure', 728)]

From this query, we can see that Disneyland has over 14,000 wait times, while Disney California Adventure only has a little over 700 wait times. What this means is that since there are more Disneyland records than Disney California Adventure records, the Disney records should be more closely analyzed until more California Adventure records are collected. However, if both Disneyland and Disney California Adventure have overlapping dates, then they could be analyzed and compared with each other. Therefore, I would recommend returning the Disneyland wait times in order to perform average, minimum, and maximum, and to collect more data for Disney California Adventures so that they could be compared later.

##### Query 3: Determining Average Wait Time for Disneyland

The purpose of this query is to identify what the average wait time for Disneyland's rides are. This data will allow us to see which rides have times that are greater than or less than this amount, which may effect how Disneyland chooses to market their rides.

In [35]:
%%sql
SELECT 
    ROUND(AVG(wait_time), 2) AS avg_disneyland_wt
FROM parks p
JOIN rides r 
    ON p.park_id = r.park_id
JOIN ride_wait_times rwt 
    ON r.id = rwt.id
WHERE p.park_id = 16
    AND rwt.is_open = 1
    AND (rwt.last_updated BETWEEN '2022-04-23' AND '2022-05-04')
   

 * mysql://admin:***@lmu-dev-02.cmrqhkyehmsg.us-west-1.rds.amazonaws.com/sql_project
1 rows affected.


[(Decimal('25.15'),)]

From this query, we can see that the average wait time for rides in Disneyland is approximately 25.15 minutes. This is important to understand because we can compare rides to see which rides have a longer or shorter wait than average, and by how much. However, because this average is around 25 minutes, which may cause disatisfaction with park attendees, I think that Disneyland should determine which rides have less than average wait times and increase marketing for those rides. By encouraging attendees to go to rides with smaller wait times, this may decrease the wait times of larger rides. Alternatively, Disney could increase advertising spending for their Genie Plus service due to larger wait times, which may increase revenue but potentially cause lines to become longer.

##### Query 4: Finding Average Wait Times for Disneyland Rides

In this query, we use the data collected over the past 11 days and perform an average in order to determine the average wait time per ride in the Disneyland Park. Having this data will allow us to compare wait times to the average later.

In [36]:
%%sql
SELECT  
	r.name,
	ROUND(AVG(rwt.wait_time), 2) AS avg_wait_time_when_open
FROM ride_wait_times rwt
JOIN rides r 
	ON rwt.id = r.id
JOIN parks p
    ON r.park_id = p.park_id
WHERE (last_updated BETWEEN '2022-04-23' AND '2022-05-04')
	AND is_open = 1
    AND r.park_id = 16
GROUP BY rwt.id;

 * mysql://admin:***@lmu-dev-02.cmrqhkyehmsg.us-west-1.rds.amazonaws.com/sql_project
33 rows affected.


[('Enchanted Tiki Room', Decimal('0.00')),
 ('Indiana Jones Adventure\xa0', Decimal('45.25')),
 ('Jungle Cruise', Decimal('27.17')),
 ('Splash Mountain', Decimal('39.10')),
 ('The Many Adventures of Winnie the Pooh', Decimal('15.36')),
 ('Alice in Wonderland', Decimal('32.90')),
 ('Dumbo the Flying Elephant', Decimal('30.19')),
 ("it's a small world", Decimal('19.26')),
 ('King Arthur Carrousel', Decimal('12.75')),
 ('Mad Tea Party\xa0', Decimal('15.25')),
 ('Matterhorn Bobsleds', Decimal('39.80')),
 ("Mr. Toad's Wild Ride\xa0", Decimal('20.72')),
 ("Peter Pan's Flight\xa0", Decimal('36.99')),
 ("Pinocchio's Daring Journey", Decimal('18.04')),
 ("Snow White's Scary Adventures", Decimal('23.16')),
 ('Storybook Land Canal Boats', Decimal('21.67')),
 ('Mark Twain Riverboat', Decimal('0.00')),
 ('Disneyland Railroad - Main Street, U.S.A.', Decimal('24.40')),
 ('The\xa0Disneyland Story\xa0presenting Great\xa0Moments with Mr. Lincoln', Decimal('0.00')),
 ('Haunted Mansion', Decimal('32.35')),
 ('Millennium Falcon: Smugglers Run', Decimal('36.85')),
 ('Star Wars: Rise of the Resistance', Decimal('70.64')),
 ('Astro Orbitor', Decimal('25.63')),
 ('Autopia', Decimal('21.74')),
 ('Buzz Lightyear Astro Blasters\xa0', Decimal('26.13')),
 ('Disneyland Monorail', Decimal('6.74')),
 ('Space Mountain', Decimal('58.49')),
 ('Star Tours\xa0- The Adventures Continue', Decimal('23.80')),
 ('Casey Jr. Circus Train', Decimal('19.24')),
 ('Big Thunder Mountain Railroad', Decimal('37.99')),
 ("Pirate's Lair on Tom Sawyer Island", Decimal('4.12')),
 ('Sailing Ship Columbia', Decimal('0.00')),
 ("Davy Crockett's Explorer Canoes", Decimal('10.45'))]

For this query, we see that there are many wait times that are between 20 and 40 minutes, which would explain why the average is around 25 minutes above. However, there are some outliers, with one wait time being close to 70 minutes and others being closer to zero. For future queries, the data should be sorted and compared to the average, which will allow Disney to better understand the rides they need to market more.

##### Query 5: Determining Rides That Have Been Closed in The Past 10 Days

For this query, we will sum the number of times a ride has been open; we will then filter the results so only the rides that have not been open show. This will allow us to see the rides that Disney should avoid marketing for the time being.

In [37]:
%%sql
SELECT 
    r.id,
    r.name AS ride_name,
    SUM(rwt.is_open) AS times_open
FROM ride_wait_times rwt
JOIN rides r 
    ON rwt.id = r.id
JOIN parks p
    ON r.park_id = p.park_id
WHERE (last_updated BETWEEN '2022-04-23' AND '2022-05-04')
    AND r.park_id = 16
GROUP BY rwt.id
HAVING SUM(rwt.is_open) = 0
ORDER BY SUM(rwt.is_open) DESC;

 * mysql://admin:***@lmu-dev-02.cmrqhkyehmsg.us-west-1.rds.amazonaws.com/sql_project
18 rows affected.


[(324, "Gadget's Go Coaster", 0.0),
 (2520, 'Star Wars Launch Bay', 0.0),
 (276, 'Finding Nemo Submarine Voyage', 0.0),
 (6712, "Star Wars: Galaxy's Edge", 0.0),
 (289, 'Pirates of the Caribbean\xa0', 0.0),
 (332, "Roger Rabbit's Car Toon Spin", 0.0),
 (684, "Minnie's House", 0.0),
 (327, "Mickey's House and Meet Mickey", 0.0),
 (680, "Goofy's Playhouse", 0.0),
 (689, "Tarzan's Treehouse", 0.0),
 (677, "Donald's Boat", 0.0),
 (673, "Chip 'n Dale Treehouse", 0.0),
 (695, 'The Disney Gallery', 0.0),
 (691, 'Main Street Vehicles', 0.0),
 (686, 'Main Street Cinema', 0.0),
 (6114, 'Fortune Tellers', 0.0),
 (679, "Frontierland Shootin' Exposition", 0.0),
 (687, 'Sleeping Beauty Castle Walkthrough', 0.0)]

For this query, we can see that there are around 18 rides that are not open, such as Pirates of the Carribean, Star Wars: Galaxy's Edge, and Tarzan's Treehouse, to name a few. Therefore, I would recommend that Disney limit its marketing and advertising for these rides and increase advertising for open rides in order to increase excitement about open rides and prevent disappointment for rides that are closed.

### Primary And Supplementary Questions

##### Primary Question: How Can Disney Reduce Wait Times for Its More Popular Rides?

The purpose of this query is to determine the average wait times for each of the rides in the past 11 days (when the rides were open), and then compare the wait times to Disneyland's average wait time per ride. This query is important because it allows Disney to identify which rides are more popular and have longer wait times. With an understanding of how each ride compares to the average, Disney can boost marketing efforts for their rides with lower wait times; this would increase wait times for rides below average and decrease rides with wait times above average.

In [38]:
%%sql
SELECT  
	r.name AS ride_name,
	ROUND(AVG(rwt.wait_time), 2) AS avg_wait_time_when_open,
    (CASE WHEN ROUND(AVG(rwt.wait_time), 2) >
    (
        SELECT 
            ROUND(AVG(wait_time), 2) AS avg_disneyland_wt
        FROM parks p
        JOIN rides r 
            ON p.park_id = r.park_id
        JOIN ride_wait_times rwt 
            ON r.id = rwt.id
        WHERE p.park_id = 16
            AND rwt.is_open = 1
    ) THEN 'Yes' ELSE 'No' END) AS greater_than_disneyland_avg
FROM ride_wait_times rwt
JOIN rides r 
	ON rwt.id = r.id
JOIN parks p
    ON r.park_id = p.park_id
WHERE (last_updated BETWEEN '2022-04-23' AND '2022-05-04')
	AND is_open = 1
    AND r.park_id = 16
GROUP BY rwt.id
ORDER BY AVG(rwt.wait_time) DESC;

 * mysql://admin:***@lmu-dev-02.cmrqhkyehmsg.us-west-1.rds.amazonaws.com/sql_project
33 rows affected.


[('Star Wars: Rise of the Resistance', Decimal('70.64'), 'Yes'),
 ('Space Mountain', Decimal('58.49'), 'Yes'),
 ('Indiana Jones Adventure\xa0', Decimal('45.25'), 'Yes'),
 ('Matterhorn Bobsleds', Decimal('39.80'), 'Yes'),
 ('Splash Mountain', Decimal('39.10'), 'Yes'),
 ('Big Thunder Mountain Railroad', Decimal('37.99'), 'Yes'),
 ("Peter Pan's Flight\xa0", Decimal('36.99'), 'Yes'),
 ('Millennium Falcon: Smugglers Run', Decimal('36.85'), 'Yes'),
 ('Alice in Wonderland', Decimal('32.90'), 'Yes'),
 ('Haunted Mansion', Decimal('32.35'), 'Yes'),
 ('Dumbo the Flying Elephant', Decimal('30.19'), 'Yes'),
 ('Jungle Cruise', Decimal('27.17'), 'Yes'),
 ('Buzz Lightyear Astro Blasters\xa0', Decimal('26.13'), 'Yes'),
 ('Astro Orbitor', Decimal('25.63'), 'Yes'),
 ('Disneyland Railroad - Main Street, U.S.A.', Decimal('24.40'), 'No'),
 ('Star Tours\xa0- The Adventures Continue', Decimal('23.80'), 'No'),
 ("Snow White's Scary Adventures", Decimal('23.16'), 'No'),
 ('Autopia', Decimal('21.74'), 'No'),
 ('Storybook Land Canal Boats', Decimal('21.67'), 'No'),
 ("Mr. Toad's Wild Ride\xa0", Decimal('20.72'), 'No'),
 ("it's a small world", Decimal('19.26'), 'No'),
 ('Casey Jr. Circus Train', Decimal('19.24'), 'No'),
 ("Pinocchio's Daring Journey", Decimal('18.04'), 'No'),
 ('The Many Adventures of Winnie the Pooh', Decimal('15.36'), 'No'),
 ('Mad Tea Party\xa0', Decimal('15.25'), 'No'),
 ('King Arthur Carrousel', Decimal('12.75'), 'No'),
 ("Davy Crockett's Explorer Canoes", Decimal('10.45'), 'No'),
 ('Disneyland Monorail', Decimal('6.74'), 'No'),
 ("Pirate's Lair on Tom Sawyer Island", Decimal('4.12'), 'No'),
 ('The\xa0Disneyland Story\xa0presenting Great\xa0Moments with Mr. Lincoln', Decimal('0.00'), 'No'),
 ('Mark Twain Riverboat', Decimal('0.00'), 'No'),
 ('Sailing Ship Columbia', Decimal('0.00'), 'No'),
 ('Enchanted Tiki Room', Decimal('0.00'), 'No')]

From this data, we are able to see that the rides with the longest average wait times (while open) are Star Wars: Rise of the Resistance, Space Mountain, Indiana Jones Adventure, and Matterhorn Bobsleds, while the rides with the shortest wait times are The Disneyland Story presenting Great Moments with Mr. Lincoln, Mark Twain Riverboat, Sailing Ship Columbia, and Enchanted Tiki Room. In order to reduce the wait times for rides with the longest average wait time, Disney should look at rides with shorter wait times that are close by and increase marketing for those rides. For example, because Star Wars: Rise of the Resistance has the longest wait, Disney should increase its advertising for Millenium Falcon: Smugglers Run; even though both have above average wait times, increasing marketing efforts for the Millenium Falcon ride may create more excitement for that ride, potentially decreasing the wait time for Rise of the Resistance. Similarly, with Space Mountain, Disney could increase marketing for nearby rides that have below average wait times, like Star Tours - The Adventures Continue, in order to elongate the wait time for the shorter rides and shorten the wait time for the longer wait time rides. Therefore, Disney should look at the average wait times for rides and compare them to the average; they should then use this information to market rides with shorter wait times in order to create more excitement for those rides, potentially decreasing wait times for more popular rides.

##### Supplementary Question: Which Rides Have Been Open Most Often In The Past 11 Days?

Now that we know the average wait time for rides while they are open, this query will "fill in the gaps" by identifying which rides are open most often. By identifying and ranking the rides that are open most often, Disney can further adjust its marketing strategy by advertising rides that are open more often; this will prevent customers for feeling frustrated or upset if a ride advertised as open ended up being closed.

In [52]:
%%sql
WITH time_open AS (
    SELECT 
        r.id,
        r.name AS ride_name,
        ROUND(SUM(CASE WHEN is_open = 1 THEN is_open ELSE 0 END) / 12, 2) AS hours_open,
        ROUND(((SUM(CASE WHEN is_open = 1 THEN is_open ELSE 0 END) / 12) / 16)*100, 2) AS open_rate
    FROM ride_wait_times rwt
    JOIN rides r 
        ON rwt.id = r.id
    JOIN parks p
        ON r.park_id = p.park_id
    WHERE last_updated >= '2022-04-23'
        AND r.park_id = 16
    GROUP BY rwt.id
    ORDER BY SUM(rwt.is_open) DESC
)
SELECT
    DENSE_RANK() OVER (
        ORDER BY open_rate DESC
    ) AS percent_of_time_open,
    ride_name,
    open_rate
FROM time_open
GROUP BY id;

 * mysql://admin:***@lmu-dev-02.cmrqhkyehmsg.us-west-1.rds.amazonaws.com/sql_project
51 rows affected.


[(1, 'Star Tours\xa0- The Adventures Continue', 98.96),
 (1, 'Millennium Falcon: Smugglers Run', 98.96),
 (2, 'Buzz Lightyear Astro Blasters\xa0', 97.92),
 (2, 'Astro Orbitor', 97.92),
 (2, 'Mad Tea Party\xa0', 97.92),
 (3, 'Enchanted Tiki Room', 97.4),
 (3, 'Haunted Mansion', 97.4),
 (3, 'Autopia', 97.4),
 (4, 'Jungle Cruise', 96.88),
 (4, 'Dumbo the Flying Elephant', 96.88),
 (4, 'Storybook Land Canal Boats', 96.88),
 (5, "it's a small world", 94.27),
 (6, 'Splash Mountain', 93.23),
 (7, 'Disneyland Railroad - Main Street, U.S.A.', 92.71),
 (7, 'Matterhorn Bobsleds', 92.71),
 (8, 'King Arthur Carrousel', 91.67),
 (9, "Mr. Toad's Wild Ride\xa0", 89.06),
 (9, "Pinocchio's Daring Journey", 89.06),
 (9, "Snow White's Scary Adventures", 89.06),
 (9, 'The Many Adventures of Winnie the Pooh', 89.06),
 (10, "Peter Pan's Flight\xa0", 88.54),
 (11, 'Indiana Jones Adventure\xa0', 88.02),
 (12, 'Big Thunder Mountain Railroad', 87.5),
 (13, 'Space Mountain', 85.42),
 (14, 'Alice in Wonderland', 83.85),
 (15, 'Casey Jr. Circus Train', 81.77),
 (16, 'The\xa0Disneyland Story\xa0presenting Great\xa0Moments with Mr. Lincoln', 80.73),
 (17, 'Star Wars: Rise of the Resistance', 70.31),
 (18, 'Mark Twain Riverboat', 66.15),
 (19, 'Disneyland Monorail', 57.81),
 (20, "Pirate's Lair on Tom Sawyer Island", 47.4),
 (21, 'Sailing Ship Columbia', 41.15),
 (22, "Davy Crockett's Explorer Canoes", 40.62),
 (23, "Tarzan's Treehouse", 0.0),
 (23, 'Finding Nemo Submarine Voyage', 0.0),
 (23, 'Star Wars Launch Bay', 0.0),
 (23, "Star Wars: Galaxy's Edge", 0.0),
 (23, 'Sleeping Beauty Castle Walkthrough', 0.0),
 (23, 'Pirates of the Caribbean\xa0', 0.0),
 (23, "Roger Rabbit's Car Toon Spin", 0.0),
 (23, "Minnie's House", 0.0),
 (23, "Mickey's House and Meet Mickey", 0.0),
 (23, "Goofy's Playhouse", 0.0),
 (23, "Gadget's Go Coaster", 0.0),
 (23, "Donald's Boat", 0.0),
 (23, "Chip 'n Dale Treehouse", 0.0),
 (23, 'The Disney Gallery', 0.0),
 (23, 'Main Street Vehicles', 0.0),
 (23, 'Main Street Cinema', 0.0),
 (23, 'Fortune Tellers', 0.0),
 (23, "Frontierland Shootin' Exposition", 0.0)]

From this query, we can see that Star Tours - The Adventures Continue, Millenium Falcon: Smugglers Run, Buzz Lightyear Astro Blasters, Astro Orbitor, and Mad Tea Party are among the rides that have been open most often, being open for more than 97% of the time the past 11 days. On the other hand, we can see that there are many rides that have not been open hardly at all in the past 11 days or so, such as Pirates of the Caribbean, Star Wars: Galaxy's Edge, and Star Wars Launch Bay, of several. With the rates that rides are open and the average wait times for open rides, Disney can better identify and select the best rides to advertise in order to reduce wait times for the most popular rides. For instance, in order to reduce the wait time for Star Wars: Rise of the Resistance, the Disney should advertise the Millenium Falcon: Smugglers Run ride not only because the wait time is lower than 40 minutes (which is over 30 minutes shorter than Rise of the Resistance) but also because according to this query, the ride is open around 99% of the time, meaning that it has been open nearly all hours that the park has been open in the past 11 days. In addition, if Disney were looking to reduce wait times for Space Mountain, this query would demonstrate that Buzz Lightyear Astro Blasters and Astro Orbiter have both been open around 98% of the time in the past 11 days, meaning that they would be reliable rides to market. Therefore, I would recommend that Disney look at both the wait times and the amount of time that the rides are open in order to advertise rides with lower wait times and high open rates, potentially lowering the wait time for Disneyland's more popular rides.

##### Supplementary Question: What Are Longest and Shortest Wait Times For The Rides That Have Been Open?

The purpose of this query would be to identify the maximum and minimum wait times in the past 11 days. This query is important because it allows Disney to understand the ride wait time extremes and the range between them. If rides have ranges that are greater than 60 minutes, then Disney should focus on marketing rides with lower wait times in order to ensure that maximum wait times can be reduced.

In [53]:
%%sql
CREATE OR REPLACE VIEW max_min_wait_times AS 
    SELECT 
        r.id,
        r.name AS ride_name,
        MAX(rwt.wait_time) AS longest_wait_time,
        MIN(rwt.wait_time) AS shortest_wait_time,
        (CASE WHEN (MAX(rwt.wait_time) - MIN(rwt.wait_time)) >= 60 THEN 'Yes' ELSE 'No' END)
        AS wt_difference_greater_than_60
    FROM ride_wait_times rwt
    JOIN rides r 
        ON rwt.id = r.id
    JOIN parks p
        ON r.park_id = p.park_id
    WHERE last_updated BETWEEN '2022-04-23' AND '2022-05-04'
        AND r.park_id = 16
        AND is_open = 1
    GROUP BY rwt.id
    ORDER BY longest_wait_time DESC;

SELECT *
FROM max_min_wait_times
ORDER BY 
    longest_wait_time DESC, 
    shortest_wait_time DESC,
    wt_difference_greater_than_60;

 * mysql://admin:***@lmu-dev-02.cmrqhkyehmsg.us-west-1.rds.amazonaws.com/sql_project
0 rows affected.
33 rows affected.


[(6340, 'Star Wars: Rise of the Resistance', 180, 10, 'Yes'),
 (284, 'Space Mountain', 150, 10, 'Yes'),
 (297, 'Splash Mountain', 135, 5, 'Yes'),
 (279, 'Matterhorn Bobsleds', 95, 10, 'Yes'),
 (326, 'Indiana Jones Adventure\xa0', 90, 10, 'Yes'),
 (6339, 'Millennium Falcon: Smugglers Run', 85, 5, 'Yes'),
 (325, 'Haunted Mansion', 80, 5, 'Yes'),
 (286, 'Star Tours\xa0- The Adventures Continue', 70, 5, 'Yes'),
 (323, 'Big Thunder Mountain Railroad', 65, 5, 'Yes'),
 (281, "Peter Pan's Flight\xa0", 60, 5, 'No'),
 (273, 'Buzz Lightyear Astro Blasters\xa0', 60, 5, 'No'),
 (275, 'Dumbo the Flying Elephant', 60, 5, 'No'),
 (287, 'Astro Orbitor', 60, 5, 'No'),
 (296, 'Jungle Cruise', 60, 0, 'Yes'),
 (285, 'Alice in Wonderland', 60, 0, 'Yes'),
 (317, 'Autopia', 55, 5, 'No'),
 (280, "Mr. Toad's Wild Ride\xa0", 55, 5, 'No'),
 (307, "it's a small world", 50, 5, 'No'),
 (305, 'Storybook Land Canal Boats', 50, 0, 'No'),
 (674, 'Disneyland Railroad - Main Street, U.S.A.', 50, 0, 'No'),
 (283, "Snow White's Scary Adventures", 45, 0, 'No'),
 (282, "Pinocchio's Daring Journey", 40, 5, 'No'),
 (306, 'The Many Adventures of Winnie the Pooh', 40, 0, 'No'),
 (277, 'King Arthur Carrousel', 35, 5, 'No'),
 (303, 'Casey Jr. Circus Train', 35, 0, 'No'),
 (278, 'Mad Tea Party\xa0', 35, 0, 'No'),
 (274, 'Disneyland Monorail', 30, 0, 'No'),
 (304, "Davy Crockett's Explorer Canoes", 25, 0, 'No'),
 (331, "Pirate's Lair on Tom Sawyer Island", 5, 0, 'No'),
 (690, 'The\xa0Disneyland Story\xa0presenting Great\xa0Moments with Mr. Lincoln', 0, 0, 'No'),
 (456, 'Mark Twain Riverboat', 0, 0, 'No'),
 (328, 'Sailing Ship Columbia', 0, 0, 'No'),
 (288, 'Enchanted Tiki Room', 0, 0, 'No')]

From this query, we can understand that the majority of rides (while they are open) have a wait time difference less than 60 minutes, or an hour. However, there are multiple rides that have very high maximums wait times, such as Star Wars: Rise of the Resistance with 180 minutes, Space Mountain with 150 minutes, and Splash Mountain with 135 minutes, to name a few. Ultimately, for all rides that have a wait time difference greater than 60, Disney needs to identify rides that have a smaller wait time difference in order reduce the more popular rides' wait times. For instance, in the case of the Matterhorn, Disney could consider marketing rides like Mad Tea Party, Snow White's Scary Adventures, and Mr. Toad's Wild Ride. Ultimately, I would recommend that Disney use Average Wait Time, Open Rate, and the difference between maximum and minimum wait times (and whether that number is greater than 60) in order to determine rides with low wait times that are close to the most popular rides. Then, Disney could use some of its marketing resources in order to advertise rides that have lower wait times, higher open rates, and smaller differences between their maximum and minimum times. This will allow Disney to increase popularity for their less popular rides, effectively lowering the wait times for their most popular rides and making them more accessible for park attendees.
    